src/test/java/com/rog/EShop/controllers/KeycloakControllerTest.java

```diff
@@ -0,0 +1,47 @@
+package com.rog.EShop.controllers;
+
+import com.rog.EShop.dto.TokenDto;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+
+@SpringBootTest
+class KeycloakControllerTest {
+    @Autowired
+    private WebApplicationContext wac;
+    private MockMvc mockMvc;
+
+    @BeforeEach
+    public void setUp() {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(this.wac).build();
+    }
+
+    @Test
+    void getToken() throws Exception {
+        TokenDto tokenDto = new TokenDto();
+        tokenDto.setUsername("user5");
+        tokenDto.setPassword("user5");
+        tokenDto.setClientId("estore-api");
+        tokenDto.setClientSecret("2iwoTq3NgMMj0CRHIjzWbZ7GRLj5mLEk");
```

Comment:
`clientId` and `clientSecret` must not be exposed in code. Use Configuration Properties to create your custom properties.